src/components/Compiler.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+class Compiler extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            dir: '',
+            text: ':/',
+            lastWord: '',
+            constTextLength: 2,
+            cursorVisible: true
+        };
+        this.cursorChar = '|';
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+        this.KEYCODE = {
+            backspace : 8,
+            enter: 13,
+            space: 32
+        };
+    }
+
+    changeCursorVisibility() {
+        this.setState({ cursorVisible: !this.state.cursorVisible });
+    }
+
+    handleKeyDown(event) {
+        let updatedText;
```

Comment:
there is a lot of `this.state` in your code, just move it to own variables, it would look a lot cleaner. Smth like:
```
const { constTextLength, text, lastWord } = this.state;
```